kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -138,17 +139,19 @@ private PartitionedTableId getRecordTable(SinkRecord record) {
     TableId baseTableId = topicsToBaseTableIds.get(record.topic());
 
     PartitionedTableId.Builder builder = new PartitionedTableId.Builder(baseTableId);
-    if (useMessageTimeDatePartitioning) {
-      if (record.timestampType() == TimestampType.NO_TIMESTAMP_TYPE) {
-        throw new ConnectException(
-            "Message has no timestamp type, cannot use message timestamp to partition.");
-      }
 
-      builder.setDayPartition(record.timestamp());
-    } else {
-      builder.setDayPartitionForNow();
-    }
+    if (explicitPartitioning) {
```

Comment:
Message time partitioning is also explicit partitioning (as we are defining the partition with the records). This config name name along with `useMessageTimeDatePartitioning` looks conflicting. Also, it's not following the pattern for the other config parameters (i.e. `useXxx`).

Reply:
No suggestion for alternative name?
How’s about useExplicitIngestionTimePartitionDecoration?

Reply:
That's too verbose. Another name could be `usePartitionDecorator` but that would mean this would become the exact replica of https://github.com/wepay/kafka-connect-bigquery/pull/229 if it's not the replica already.